restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/entities/SmsMessageFilter.java

```diff
@@ -19,38 +19,48 @@
  */
 package org.restcomm.connect.dao.entities;
 
+import org.restcomm.connect.commons.annotations.concurrency.Immutable;
+
+import java.text.DateFormat;
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.Date;
 import java.util.List;
 
-import org.restcomm.connect.commons.annotations.concurrency.Immutable;
-
 /**
  * @author <a href="mailto:n.congvu@gmail.com">vunguyen</a>
+ * @author Henrique Rosa (henrique.rosa@telestax.com)
  */
 
 @Immutable
 public class SmsMessageFilter {
 
-    private final String accountSid;
-    private final List<String> accountSidSet; // if not-null we need the cdrs that belong to several accounts
-    private final String recipient;
-    private final String sender;
-    private final Date startTime;  // to initialize it pass string arguments with  yyyy-MM-dd format
-    private final Date endTime;
-    private final String body;
-    private final Integer limit;
-    private final Integer offset;
-    private final String instanceid;
+    private static final DateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
+
+    private String accountSid;
+    private List<String> accountSidSet; // if not-null we need the cdrs that belong to several accounts
+    private String recipient;
+    private String sender;
+    private String startTime;  // to initialize it pass string arguments with  yyyy-MM-dd format
```

Comment:
what was the reason to change the type from Date to String

Reply:
As far as the filter is concerned, the date is a String with some date-like pattern.
If you check the builder, it accepts a Date which is internally parsed to the desired format. IMO much cleaner and avoids parsing exception when building these objects.